### PR TITLE
Feature/deferred hypha payment

### PR DESCRIPTION
**Pull request by n13**
HIP 1 - deferred payouts of Hypha into the costake contract

https://github.com/hypha-dao/dao-contracts/issues/168

Changes

- [x] Hypha payments are deferred
- [x] Removed pausing of payouts
- [x] New setting for cosale contract name `#define HYPHA_COSALE_CONTRACT "hypha_cosale_contract"`

Still to do
set the contract name setting 
set setting "hypha_cosale_contract" to "costak.hypha"
set setting "hypha_usd_value" to 50 cents 

The change to hypha payer is that it now pays the amount fo the costak.hypha contract, and that it then creates a lock for the user. The expiry date on the lock is set to 3 years but we don't use that right now in the cosale contract.

### CODE REVIEW

File: hydra.yml [hunk 1]
```diff
@@ -32,4 +32,8 @@ contracts:
     abi: dao-go/artifacts/voice/voice.abi
     name: voice
     wasm: dao-go/artifacts/voice/voice.wasm
+  cosale:
+    abi: dao-go/artifacts/vesting/vesting.abi
+    name: cosale
+    wasm: dao-go/artifacts/vesting/vesting.wasm
```
Comment (josejulio) [inline hydra.yml:38]: This file is missing from this PR. I think you have to force add it.

Comment (n13) [reply hydra.yml:38]: Hmm I did force add it, but since wasm is in gitignore, weird things may happen

Comment (josejulio) [reply hydra.yml:38]: Could you try with `git add --force dao-go/artifacts/vesting/vesting.wasm` ? otherwise we need to clean up the .gitignore

Comment (n13) [reply hydra.yml:38]: Ok did it again

Comment (n13) [reply hydra.yml:38]: I think one of them was force added, but not the other, now both are force added

File: src/payers/hypha_payer.cpp [hunk 2]
```diff
@@ -18,9 +20,19 @@ namespace hypha
...
+
+        eosio::action(
+            eosio::permission_level{m_dao.get_self(), name("active")},
+            m_dao.getSettingOrFail<eosio::name>(HYPHA_COSALE_CONTRACT), name("createlock"),
+            std::make_tuple(m_dao.get_self(),
+                            recipient,
+                            quantity,
+                            memo,
+                            eosio::time_point(eosio::current_time_point().time_since_epoch() + EXPIRY_MILLISECONDS)))
+            .send();
```
Comment (Gerard097) [inline src/payers/hypha_payer.cpp:35]: time_since_epoch() returns in microseconds, although we are adding to milliseconds (EXPIRY_MILLISECONDS). We should convert to EXPIRY_MILLISECONDS to microseconds as well to get the right amount.

**Review by Gerard097 — CHANGES_REQUESTED** [review]

File: hydra.yml (hunk 1 above)
Comment (josejulio) [reply hydra.yml:38]: I think you forgot to push it - I can't see the vesting.wasm file in the PR

File: src/payers/hypha_payer.cpp (hunk 2 above)
Comment (n13) [reply src/payers/hypha_payer.cpp:35]: Oh gosh, I confused the two. Thanks!

**Review by Gerard097 — APPROVED** [review]
Looks good!

**Pull request merged**